controller/hybridgateway/converter/http_route.go

```diff
@@ -266,15 +266,23 @@ func (c *httpRouteConverter) addHostnames(ctx context.Context) error {
 					// This listener doesn't match the section reference, skip it
 					continue
 				}
+				if listener.Port != *pRef.Port {
```

Comment:
Are we sure that `pRef.Port` won't be `nil`? If not, we can do 


```suggestion
				if listener.Port != lo.FromPtr(pRef.Port) {
```

Reply:
good catch! The port should be mandatory but using the `lo` package would be safer (moreover is better also for consistency as we use the `lo` package extensively).